### PR TITLE
Filter kernels with build errors

### PR DESCRIPTION
**Pull request by bstefanuk**
Simple refactor:
1. Add a function in TensileCreateLibrary.py to filter kernels that emit and error when building (relates to kernel writing logic).
2. Add **isort** to automatically sort import for select files—reduces merge conflicts.
3. Update `printExit` to display red.

**Testing**

Unit tests are added for the new function and existing related functions.

**Notable changes**

1. The `TensileCreateLibrary` import in *test_TensileCreateLibrary* is renamed to `tcl`. So all functions should be accessed like `tcl.functionName(...)`.
2. An import sorting check has been added to CI, so make sure to run `tox run -m static` prior to committing to ensure formatting, linting, and import sorting are all complete.

### CODE REVIEW

File: Tensile/TensileCreateLibrary.py [hunk 1]
```diff
@@ -751,7 +764,7 @@ def success(kernel):
         (err, src) = ko.getSourceFileString()
         kernelSourceFile.write(src)
         if err:
-            print("*** warning: invalid kernel#%u" % kernelName)
+            printWarning("*** warning: invalid kernel#%u" % kernelName)
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:767]: Is it redundant to have warning in the string?

Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:767]: Yes it is, will remove, good catch.

File: Tensile/Tests/unit/test_TensileCreateLibrary.py [hunk 2]
```diff
@@ -626,9 +606,118 @@ def test_filterProcessingErrors(setupSolutionsAndKernels):
...
+
+    assert (
+        kernelFiles[0] == "no-commit-kernel-build-files/Kernels.cpp"
+    ), "Cpp source file doesn't match"
+    assert (
+        kernelsWithBuildErrors == expectedWithBuildErrors
+    ), "Kernels with build errors don't match expectation"
+
+
+def test_filterBuildErrors(setupSolutionsAndKernels):
```
Comment (ellosel) [inline Tensile/Tests/unit/test_TensileCreateLibrary.py:683]: Use mocks for testing.

Comment (bstefanuk) [reply Tensile/Tests/unit/test_TensileCreateLibrary.py]: Done.

File: Tensile/TensileCreateLibrary.py [hunk 3]
```diff
@@ -85,7 +80,7 @@ def processKernelSource(kernel, kernelWriterSource, kernelWriterAssembly):
     Returns (error, source, header, kernelName).
     """
     try:
-        kernelWriter = (
+        kernelWriter: KernelWriterAssembly | KernelWriterSource = (
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:83]: Can't use `|`...

File: Tensile/TensileCreateLibrary.py [hunk 4]
```diff
@@ -642,6 +638,38 @@ def filterProcessingErrors(
     return keepKernels, keepSolutions, keepResults
 
 
+def filterBuildErrors(
+    kernels: List[Solution],
+    kernelsWithBuildErrors: Dict[str, int],
+    writerSelectionFn: Callable[[str], KernelWriterSource | KernelWriterAssembly],
```
Comment (ellosel) [inline Tensile/TensileCreateLibrary.py:644]: Use `Optional`

File: Tensile/TensileCreateLibrary.py (hunk 3 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:83]: Removed, this type hint isn't very valuable anyways.

File: Tensile/TensileCreateLibrary.py (hunk 4 above)
Comment (bstefanuk) [reply Tensile/TensileCreateLibrary.py:644]: Adding `Union` instead of `Optional` which is equivalent to `Union[YourType, None]`

**Review by ellosel — APPROVED** [review]

**Pull request merged**